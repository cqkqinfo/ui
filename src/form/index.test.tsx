import React from 'react';
import Form from './index';
import FormItem from '../form-item';
import { render } from '@testing-library/react';

describe('表单', () => {
  it('渲染', () => {
    const { baseElement } = render(
      <Form>
        <FormItem name={'a'} label={'姓名'} />
        <FormItem
          name={'b'}
          rules={[{ required: true, type: 'phone' }]}
          label={'手机号'}
        />
      </Form>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('cell模式', () => {
    const { baseElement } = render(
      <Form cell>
        <FormItem name={'a'} label={'姓名'} />
        <FormItem
          name={'b'}
          rules={[{ required: true, type: 'phone' }]}
          label={'手机号'}
        />
      </Form>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('card模式', () => {
    const { baseElement } = render(
      <Form card>
        <FormItem name={'a'} label={'姓名'} />
        <FormItem
          name={'b'}
          rules={[{ required: true, type: 'phone' }]}
          label={'手机号'}
        />
      </Form>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('两端对齐', () => {
    const { baseElement } = render(
      <Form labelWidth={'3em'}>
        <FormItem name={'a'} label={'姓名'} />
        <FormItem
          name={'b'}
          rules={[{ required: true, type: 'phone' }]}
          label={'手机号'}
        />
      </Form>,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
