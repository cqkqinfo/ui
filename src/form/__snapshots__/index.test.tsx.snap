// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`表单 card模式 1`] = `
<body>
  <div>
    <div
      class="space vertical"
      style="line-height: 1; padding: 0px 10px; align-self: stretch; box-sizing: border-box; background-color: rgb(255, 255, 255); border-radius: 10px; box-shadow: 0 0 10px rgba(39,128,217, 0.15);"
    >
      <div
        class="item card"
      >
        <div
          class="labelWrap"
        >
          <div
            class="mark"
            style="opacity: 0;"
          >
            *
          </div>
          <div
            class="label"
            style="justify-content: flex-end;"
          >
            <div>
              姓
            </div>
            <div>
              名
            </div>
          </div>
          <div
            class="colon"
          >
            :
          </div>
        </div>
        <div
          class="children"
          style="justify-content: flex-end;"
        >
          <input
            class="web input"
            placeholder="请输入姓名"
            placeholderstyle="[object Object]"
            toutiao-selection-end="999"
            toutiao-selection-start="999"
            value=""
            wechat-confirm-type="done"
            wechat-cursor-spacing="0"
            wechat-placeholder-class="input-placeholder"
            wechat-selection-end="-1"
            wechat-selection-start="-1"
          />
        </div>
        <div
          class="after"
          style="display: none;"
        />
      </div>
      <div
        class="item card"
      >
        <div
          class="labelWrap"
        >
          <div
            class="mark"
            style="opacity: 1;"
          >
            *
          </div>
          <div
            class="label"
            style="justify-content: flex-end;"
          >
            <div>
              手
            </div>
            <div>
              机
            </div>
            <div>
              号
            </div>
          </div>
          <div
            class="colon"
          >
            :
          </div>
        </div>
        <div
          class="children"
          style="justify-content: flex-end;"
        >
          <input
            class="web input"
            placeholder="请输入手机号"
            placeholderstyle="[object Object]"
            toutiao-selection-end="999"
            toutiao-selection-start="999"
            value=""
            wechat-confirm-type="done"
            wechat-cursor-spacing="0"
            wechat-placeholder-class="input-placeholder"
            wechat-selection-end="-1"
            wechat-selection-start="-1"
          />
        </div>
        <div
          class="after"
          style="display: none;"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`表单 cell模式 1`] = `
<body>
  <div>
    <div
      class="space vertical"
      style="line-height: 1; padding: 0px 10px; align-self: stretch; box-sizing: border-box; background-color: rgb(255, 255, 255); border-radius: 10px; box-shadow: 0 0 10px rgba(39,128,217, 0.15);"
    >
      <div
        class="item cell"
      >
        <div
          class="labelWrap"
        >
          <div
            class="mark"
            style="opacity: 0;"
          >
            *
          </div>
          <div
            class="label"
            style="justify-content: flex-end;"
          >
            <div>
              姓
            </div>
            <div>
              名
            </div>
          </div>
        </div>
        <div
          class="children"
          style="justify-content: flex-end;"
        >
          <input
            class="web input"
            placeholder="请输入姓名"
            placeholderstyle="[object Object]"
            toutiao-selection-end="999"
            toutiao-selection-start="999"
            value=""
            wechat-confirm-type="done"
            wechat-cursor-spacing="0"
            wechat-placeholder-class="input-placeholder"
            wechat-selection-end="-1"
            wechat-selection-start="-1"
          />
        </div>
        <div
          class="after"
          style="display: none;"
        />
      </div>
      <div
        class="item cell"
      >
        <div
          class="labelWrap"
        >
          <div
            class="mark"
            style="opacity: 1;"
          >
            *
          </div>
          <div
            class="label"
            style="justify-content: flex-end;"
          >
            <div>
              手
            </div>
            <div>
              机
            </div>
            <div>
              号
            </div>
          </div>
        </div>
        <div
          class="children"
          style="justify-content: flex-end;"
        >
          <input
            class="web input"
            placeholder="请输入手机号"
            placeholderstyle="[object Object]"
            toutiao-selection-end="999"
            toutiao-selection-start="999"
            value=""
            wechat-confirm-type="done"
            wechat-cursor-spacing="0"
            wechat-placeholder-class="input-placeholder"
            wechat-selection-end="-1"
            wechat-selection-start="-1"
          />
        </div>
        <div
          class="after"
          style="display: none;"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`表单 两端对齐 1`] = `
<body>
  <div>
    <div
      class="space vertical"
      style="line-height: 1; align-self: stretch;"
    >
      <div
        class="item"
      >
        <div
          class="labelWrap"
        >
          <div
            class="mark"
            style="opacity: 0;"
          >
            *
          </div>
          <div
            class="label"
            style="min-width: 3em; justify-content: space-between;"
          >
            <div>
              姓
            </div>
            <div>
              名
            </div>
          </div>
          <div
            class="colon"
          >
            :
          </div>
        </div>
        <div
          class="children"
          style="justify-content: flex-end;"
        >
          <input
            class="web input"
            placeholder="请输入姓名"
            placeholderstyle="[object Object]"
            toutiao-selection-end="999"
            toutiao-selection-start="999"
            value=""
            wechat-confirm-type="done"
            wechat-cursor-spacing="0"
            wechat-placeholder-class="input-placeholder"
            wechat-selection-end="-1"
            wechat-selection-start="-1"
          />
        </div>
        <div
          class="after"
          style="display: none;"
        />
      </div>
      <div
        class="item"
      >
        <div
          class="labelWrap"
        >
          <div
            class="mark"
            style="opacity: 1;"
          >
            *
          </div>
          <div
            class="label"
            style="min-width: 3em; justify-content: space-between;"
          >
            <div>
              手
            </div>
            <div>
              机
            </div>
            <div>
              号
            </div>
          </div>
          <div
            class="colon"
          >
            :
          </div>
        </div>
        <div
          class="children"
          style="justify-content: flex-end;"
        >
          <input
            class="web input"
            placeholder="请输入手机号"
            placeholderstyle="[object Object]"
            toutiao-selection-end="999"
            toutiao-selection-start="999"
            value=""
            wechat-confirm-type="done"
            wechat-cursor-spacing="0"
            wechat-placeholder-class="input-placeholder"
            wechat-selection-end="-1"
            wechat-selection-start="-1"
          />
        </div>
        <div
          class="after"
          style="display: none;"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`表单 渲染 1`] = `
<body>
  <div>
    <div
      class="space vertical"
      style="line-height: 1; align-self: stretch;"
    >
      <div
        class="item"
      >
        <div
          class="labelWrap"
        >
          <div
            class="mark"
            style="opacity: 0;"
          >
            *
          </div>
          <div
            class="label"
            style="justify-content: flex-end;"
          >
            <div>
              姓
            </div>
            <div>
              名
            </div>
          </div>
          <div
            class="colon"
          >
            :
          </div>
        </div>
        <div
          class="children"
          style="justify-content: flex-end;"
        >
          <input
            class="web input"
            placeholder="请输入姓名"
            placeholderstyle="[object Object]"
            toutiao-selection-end="999"
            toutiao-selection-start="999"
            value=""
            wechat-confirm-type="done"
            wechat-cursor-spacing="0"
            wechat-placeholder-class="input-placeholder"
            wechat-selection-end="-1"
            wechat-selection-start="-1"
          />
        </div>
        <div
          class="after"
          style="display: none;"
        />
      </div>
      <div
        class="item"
      >
        <div
          class="labelWrap"
        >
          <div
            class="mark"
            style="opacity: 1;"
          >
            *
          </div>
          <div
            class="label"
            style="justify-content: flex-end;"
          >
            <div>
              手
            </div>
            <div>
              机
            </div>
            <div>
              号
            </div>
          </div>
          <div
            class="colon"
          >
            :
          </div>
        </div>
        <div
          class="children"
          style="justify-content: flex-end;"
        >
          <input
            class="web input"
            placeholder="请输入手机号"
            placeholderstyle="[object Object]"
            toutiao-selection-end="999"
            toutiao-selection-start="999"
            value=""
            wechat-confirm-type="done"
            wechat-cursor-spacing="0"
            wechat-placeholder-class="input-placeholder"
            wechat-selection-end="-1"
            wechat-selection-start="-1"
          />
        </div>
        <div
          class="after"
          style="display: none;"
        />
      </div>
    </div>
  </div>
</body>
`;
